### ISSUE TITLE
SSL error when posting a link to a website with broken HTTPS link in meta og:image

### ISSUE THREAD

**Issue by martinvonwittich** [RocketChat/Rocket.Chat#9653]
### Description:

I just posted a link to https://www.smore.com/clippy-js into a Rockat.Chat channel. This caused a "Do you trust certificate from "DigiCert Baltimore CA-2 G2"?" error popup:

<img width="424" alt="error" src="https://user-images.githubusercontent.com/6863569/36023568-2b19cada-0d8d-11e8-9b96-036b7e70079b.png">

The cause of this error is this meta tag on the website:

```
    <meta property="og:image" content="https://clippy.js.s3.amazonaws.com/og.jpg"/>
```

The SSL certificate covers only *.s3.amazonaws.com, so clippy.js.s3.amazonaws.com is not covered.

Rocket.Chat should IMO only log errors like these in the console, and not load the preview image.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.60.3
* Operating System: Debian stretch
* Deployment Method(snap/docker/tar/etc): tar
* Number of Running Instances: 1
* DB Replicaset Oplog: Disabled
* Node Version: v8.9.3

### Steps to Reproduce:

1. Post a link to https://www.smore.com/clippy-js in a Rocket.Chat channel

### Expected behavior:

No image should be loaded, no popup error message should be displayed.

### Actual behavior:

An error message is displayed in a popup.

**Comment by martinvonwittich** [comment]
Hm, this might actually be a bug in Rocket.Chat.Electron; I can't reproduce it when opening RocketChat in a browser. Chrome will behave as I would have expected - it won't load the image, won't show an error message, and instead it will just log the error:

```
Failed to load resource: net::ERR_INSECURE_RESPONSE
```

**Comment by xbolshe** [comment]
clippy.js.s3.amazonaws.com is covered by *.s3.amazonaws.com. It seems the problem is related with an intermediate CA certificate Baltimore CyberTrust Root, which is not installed on your computer. 

Try to check: [https://baltimore-cybertrust-root.chain-demos.digicert.com/](https://baltimore-cybertrust-root.chain-demos.digicert.com/ ). 

You may download and install it in case of problem: 

[https://www.digicert.com/digicert-root-certificates.htm](https://www.digicert.com/digicert-root-certificates.htm)    

[https://dl.cacerts.digicert.com/DigiCertBaltimoreCA-2G2.crt](https://dl.cacerts.digicert.com/DigiCertBaltimoreCA-2G2.crt)

**Comment by martinvonwittich** [comment]
> clippy.js.s3.amazonaws.com is covered by *.s3.amazonaws.com.

No, it's not? Wildcard subdomains cover exactly one level, so https://js.s3.amazonaws.com would be fine (and indeed my Chrome says "secure"), while https://clippy.js.s3.amazonaws.com is not (Chrome says "NET::ERR_CERT_COMMON_NAME_INVALID").

> Try to check: https://baltimore-cybertrust-root.chain-demos.digicert.com/.

Loads fine in Chrome.

**Comment by pho3nixx** [comment]
Do/did you use the Snap package of RocketChat, @martinvonwittich?
I experienced something similar today, turns out: apparently, the Snap build ignores my system-wide installed custom root cert, while the Debian build works flawlessly without this error.

**Comment by martinvonwittich** [comment]
> Do/did you use the Snap package of RocketChat, @martinvonwittich?
> I experienced something similar today, turns out: apparently, the Snap build ignores my system-wide installed custom root cert, while the Debian build works flawlessly without this error.

You mean on the client? We migrated from Rocket.Chat to Matrix years ago, so I'm not absolutely sure how I had installed the Rocket.Chat client. I'm pretty sure though I used the Ubuntu .deb packages with the respective APT mirror, because I never liked this Snap stuff.

**Comment by pho3nixx** [comment]
Yes, I meant the client. Ok, then this seems to have another cause.
I am with you, not liking Snap either - especially after today's cert problems.
